server/repository/src/db_diesel/name_tag_row.rs

```diff
@@ -0,0 +1,125 @@
+use super::{name_tag_row::name_tag::dsl as name_tag_dsl, StorageConnection};
+
+use crate::repository_error::RepositoryError;
+
+use diesel::prelude::*;
+
+table! {
+    name_tag (id) {
+        id -> Text,
+        name -> Text,
+    }
+}
+
+#[derive(Clone, Queryable, Insertable, Debug, PartialEq, Eq, AsChangeset, Default)]
+#[table_name = "name_tag"]
+pub struct NameTagRow {
+    pub id: String,
+    pub name: String,
+}
+
+pub struct NameTagRowRepository<'a> {
+    connection: &'a StorageConnection,
+}
+
+impl<'a> NameTagRowRepository<'a> {
+    pub fn new(connection: &'a StorageConnection) -> Self {
+        NameTagRowRepository { connection }
+    }
+
+    #[cfg(feature = "postgres")]
+    pub fn upsert_one(&self, row: &NameTagRow) -> Result<(), RepositoryError> {
+        diesel::insert_into(name_tag_dsl::name_tag)
+            .values(row)
+            .on_conflict(name_tag_dsl::id)
+            .do_update()
+            .set(row)
+            .execute(&self.connection.connection)?;
+        Ok(())
+    }
+
+    #[cfg(not(feature = "postgres"))]
+    pub fn upsert_one(&self, row: &NameTagRow) -> Result<(), RepositoryError> {
+        diesel::replace_into(name_tag_dsl::name_tag)
+            .values(row)
+            .execute(&self.connection.connection)?;
+        Ok(())
+    }
+
+    pub fn find_one_by_id(&self, id: &str) -> Result<Option<NameTagRow>, RepositoryError> {
+        let result = name_tag_dsl::name_tag
+            .filter(name_tag_dsl::id.eq(id))
+            .first(&self.connection.connection)
+            .optional()?;
+        Ok(result)
+    }
+
+    pub fn delete(&self, id: &str) -> Result<(), RepositoryError> {
```

Comment:
hmm, do we actually remove name_tags in mSupply ? This would break referential integrity, or did we agree name_tags in context of programSettings would not have that constraint ? I think either way it's good to check if mSupply actually deletes name_tag

Reply:
Removed Deletes